plugins/net_plugin/net_plugin.cpp

```diff
@@ -3694,6 +3694,8 @@ namespace eosio {
       // post to dispatcher strand so that we don't have multiple threads validating the block header
       peer_dlog(this, "posting block ${n} to dispatcher strand", ("n", ptr->block_num()));
       my_impl->dispatcher.strand.post([id, c{shared_from_this()}, ptr{std::move(ptr)}, cid=connection_id]() mutable {
+         if (app().is_quiting()) // large sync span can have many of these queued up, exit quickly
```

Comment:
Why not check this before `post`?

Reply:
The only thing that happens before the post is pulling the block off the wire. The expensive part is done inside this post. So this implicit queue can become large. The check needs to be inside since that is what is queued up.